# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/xhr/XMLHttpRequestMainThread.cpp
# Commit: 17a501a19918
# Full Hash: 17a501a1991842d8488ff3e7d4611bb688abfc87
# Author: Randell Jesup <rjesup@wgate.com>
# Date: 2022-03-24 21:54:20
# Regressor Bug: 1744043
# File Overlap Count: 1
# Description:
#   Bug 1744043: Clean up nsJAR r=nika,valentin,extension-reviewers,robwu
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D132794
# ==============================================================================

diff -r daed1e9ef335 -r 17a501a19918 dom/xhr/XMLHttpRequestMainThread.cpp
--- a/dom/xhr/XMLHttpRequestMainThread.cpp	Thu Mar 24 13:45:28 2022 +0000
+++ b/dom/xhr/XMLHttpRequestMainThread.cpp	Thu Mar 24 13:54:56 2022 +0000
@@ -3962,10 +3962,9 @@
   nsresult rv;
 
   // Open Jar file to get related attributes of target file.
-  RefPtr<nsZipArchive> zip = new nsZipArchive();
-  rv = zip->OpenArchive(aJarFile);
-  if (NS_FAILED(rv)) {
-    return rv;
+  RefPtr<nsZipArchive> zip = nsZipArchive::OpenArchive(aJarFile);
+  if (!zip) {
+    return NS_ERROR_FAILURE;
   }
   nsZipItem* zipItem = zip->GetItem(aFile.get());
   if (!zipItem) {